File: Frontend/src/app/viewModels/Types/FollowType.ts
/**
 * A type class that represents a user that follows another user
 */
export class FollowType {

    /**
     * Construct's an instance of {@link FollowType}
     * 
     * @param userName user's user name.
     * @param name user's name.
     * @param follow Specifies if the user follows the user.
     * @param profilePhoto user's profile photo url
     */
    constructor(
        public userName: string,
        public name: string,
        public follow: Boolean,
        public profilePhoto: string,
    ) { }
}

File: lei_lds2122_grupo2-main/Backend/Data/ViewModels/Types/FollowType.cs
namespace Backend.Data.ViewModels.Types
{
    /// <summary>
    /// Struct that repesents a follower or following 
    /// </summary>
    public class FollowType
    {
        public string UserName { get; set; }
        public string Name { get; set; }
        public Boolean Follow { get; set; }
        public string ProfilePhoto { get; set; }

        /// <summary>
        /// Struct's constructor method
        /// </summary>
        /// <param name="userName">User's unique identifier</param>
        /// <param name="name">User's name</param>
        /// <param name="follow">Whether or not the user follows this user</param>
        /// <param name="profilePhoto">User's profile photo </param>
        public FollowType(string userName, string name, Boolean follow, string profilePhoto)
        {
            this.UserName = userName;
            this.Name = name;
            this.Follow = follow;
            this.ProfilePhoto = profilePhoto;
        }
        public FollowType()
        {
            this.UserName = null!;
            this.Name = null!;
            this.ProfilePhoto = null!;
        }
    }
}
